src/Controllers/AnnouncementController.php

```diff
@@ -112,21 +112,20 @@ public function delete($id)
         try {
             $user = auth()->user();
 
-            $announcement = Announcement::join('badaso_courses', 'badaso_courses.id', '=', 'badaso_announcements.course_id')
-                ->where('badaso_announcements.id', $id)
-                ->where(function ($query) {
-                    $query->where('badaso_announcements.created_by', auth()->user()->id)
-                        ->orWhere('badaso_courses.created_by', auth()->user()->id);
-                })
-                ->select('badaso_announcements.*')
-                ->first();
-
+            $announcement = Announcement::find($id);
             if (! $announcement) {
                 throw ValidationException::withMessages([
                     'id' => 'Announcement not found',
                 ]);
             }
 
+            $course = $announcement->course;
+            if (! ($announcement->created_by == $user->id || $course->created_by == $user->id)) {
```

Comment:
info: can also be rewritten to `if (! in_array($user->id, [$announcement->created_by, $course->created_by]))`
but it's up to you to choose the one that is more readable

Reply:
Woww TFI Aldi 🤯👍